src/main/kotlin/net/irisshaders/lilybot/extensions/util/ModUtilities.kt

```diff
@@ -36,86 +37,79 @@ class ModUtilities : Extension() {
-						return@action
-					}
-					val config = DatabaseHelper.getConfig(guild!!.id)
-					if (config == null) {
-						respond {
-							content =
-								"**Error:** Unable to access config for this guild! Please inform a member of staff"
-						}
-						return@action
-					}
-					val actionLog = guild?.getChannel(config.modActionLog) as GuildMessageChannelBehavior
-					val targetChannel = if (arguments.targetChannel == null) {
-						channel
+				val config = DatabaseHelper.getConfig(guild!!.id)!!
+				val actionLog = guild!!.getChannel(config.modActionLog) as GuildMessageChannelBehavior
+				val targetChannel =
+					if (arguments.channel != null) {
+						// This odd syntax is necessary for casting to MessageChannelBehavior
+						guild!!.getChannel(arguments.channel!!.id) as MessageChannelBehavior
 					} else {
-						guild?.getChannel(arguments.targetChannel!!.id) as MessageChannelBehavior
+						channel
 					}
 
-					try {
-						if (arguments.embedMessage) {
-							targetChannel.createEmbed {
-								color = DISCORD_BLURPLE
-								description = arguments.messageArgument
+				try {
+					if (arguments.embed) {
+						targetChannel.createEmbed {
+							color = arguments.color
+							description = arguments.message
+							if (arguments.timestamp) {
 								timestamp = Clock.System.now()
 							}
-						} else {
-							targetChannel.createMessage {
-								content = arguments.messageArgument
-							}
 						}
-					} catch (e: KtorRequestException) {
-						respond { content = "Lily does not have permission to send messages in this channel." }
-						return@action
+					} else {
+						targetChannel.createMessage {
+							content = arguments.message
+						}
 					}
+				} catch (e: KtorRequestException) {
+					respond { content = "Lily does not have permission to send messages in this channel." }
+					return@action
+				}
 
-					respond { content = "Message sent." }
-
-					actionLog.createEmbed {
-						title = "Say command used"
-						description = "`${arguments.messageArgument}`"
-						color = DISCORD_BLACK
-						timestamp = Clock.System.now()
+				respond { content = "Message sent." }
+
+				actionLog.createEmbed {
+					title = "Say command used"
+					description = "```${arguments.message}```"
+					color = DISCORD_BLACK
+					timestamp = Clock.System.now()
+					field {
+						name = "Channel:"
+						value = targetChannel.mention
+						inline = true
+					}
+					field {
+						name = "Type:"
+						value = if (arguments.embed) "Embed" else "Message"
+						inline = true
+					}
+					if (arguments.embed) {
 						field {
-							name = "Channel:"
-							value = targetChannel.mention
+							name = "Color:"
+							value = arguments.color.toString()
 							inline = true
 						}
 						field {
-							name = "Type:"
-							value = if (arguments.embedMessage) "Embed" else "Message"
+							name = "Timestamp:"
+							value = if (arguments.timestamp) "Yes" else "No"
```

Comment:
Do we really need to log this? It's not crucial information, we know when it was sent from the logging info anyway

Reply:
This doesn't log the timestamp, it logs wether or no the message had a timestamp. I personally think it's worth including just so we're not loosing info, it doesn't take much space.